src/Core/Validation/FieldValidation/DateFieldValidator.php

```diff
@@ -2,29 +2,114 @@
  *
  * Blank string values are allowed
  */
 class DateFieldValidator extends FieldValidator
 {
+    /**
+     * Minimum value as a date string
+     */
+    private ?string $minValue;
+
+    /**
+     * Minimum value as a unix timestamp
+     */
+    private ?int $minTimestamp;
+
+    /**
+     * Maximum value as a date string
+     */
+    private ?string $maxValue;
+
+    /**
+     * Maximum value as a unix timestamp
+     */
+    private ?int $maxTimestamp;
+
+    /**
+     * Converter to convert date strings to another format for display in error messages
+     *
+     * @var callable
+     */
+    private $converter;
+
+    public function __construct(
+        string $name,
+        mixed $value,
+        ?string $minValue = null,
+        ?string $maxValue = null,
+        ?callable $converter = null,
+    ) {
+        // Convert Y-m-d args to timestamps
+        // Intermiediate variables are used to prevent "must not be accessed before initialization" PHP errors
+        // when reading properties in the constructor
+        $minTimestamp = null;
+        $maxTimestamp = null;
+        if (!is_null($minValue)) {
+            $minTimestamp = $this->dateToTimestamp($minValue);
+        }
+        if (!is_null($maxValue)) {
+            $maxTimestamp = $this->dateToTimestamp($maxValue);
+        }
+        if (!is_null($minTimestamp) && !is_null($maxTimestamp) && $maxTimestamp < $minTimestamp) {
+            throw new InvalidArgumentException('maxValue cannot be less than minValue');
+        }
+        $this->minValue = $minValue;
+        $this->maxValue = $maxValue;
+        $this->minTimestamp = $minTimestamp;
+        $this->maxTimestamp = $maxTimestamp;
+        $this->converter = $converter;
+        parent::__construct($name, $value);
+    }
+
     protected function validateValue(): ValidationResult
     {
         $result = ValidationResult::create();
         // Allow empty strings
         if ($this->value === '') {
             return $result;
         }
-        // Not using symfony/validator because it was allowing d-m-Y format strings
-        $date = date_parse_from_format($this->getFormat(), $this->value ?? '');
-        if ($date === false || $date['error_count'] > 0 || $date['warning_count'] > 0) {
+        // Validate value is a valid date
+        try {
+            $timestamp = $this->dateToTimestamp($this->value ?? '');
+        } catch (Exception|Error) {
             $result->addFieldError($this->name, $this->getMessage());
+            return $result;
+        }
+        // Validate value is within range
+        if (!is_null($this->minTimestamp) && $timestamp < $this->minTimestamp) {
+            $minValue = $this->minValue;
+            if (!is_null($this->converter)) {
+                $minValue = call_user_func($this->converter, $this->minValue) ?: $this->minValue;
+            }
+            $message = _t(
+                __CLASS__ . '.TOOSMALL',
+                'Value cannot be less than {minValue}',
```

Comment:
I've swapped out the words more/less for newer/older, though I've retained the short message, because this will also be used in a programmatic context